packages/core/src/lib/actions/callback/index.ts

```diff
@@ -363,6 +366,124 @@ export async function callback(
 
       await events.signIn?.({ user, account })
 
+      return { redirect: callbackUrl, cookies }
+    } else if (provider.type === "webauthn" && method === "POST") {
+      // Get callback action from request. It should be either "authenticate" or "register"
+      const action = request.body?.action as unknown
```

Comment:
```suggestion
      const action = request.body?.action
```
we don't need type casting here right?

Reply:
without it `action` would be `any`. Though, the next line narrows the type enough, so the casting here can be safely removed 👍